### PR TITLE
Cassandra - Add username customization

### PR DESCRIPTION
**Pull request by pcman312**
Adds the ability to customize username generation for dynamic users in Cassandra.

Uses the new field `username_template` with the [go template language](https://golang.org/pkg/text/template/).

### CODE REVIEW

File: plugins/database/cassandra/connection_producer.go
Comment (catsby) [inline plugins/database/cassandra/connection_producer.go:128]: nit: could use this opportunity to change these to `fmt.Errorf("blah %w", err)` while we're here

**Review by catsby — APPROVED** [review]
LGTM 👍 minor nit/suggestion about changing from `errwrap` -> `fmt.Errorf` while we're here, but don't feel blocked by it

File: plugins/database/cassandra/connection_producer.go
Comment (pcman312) [reply plugins/database/cassandra/connection_producer.go:128]: I had held off on this because we hadn't updated `errwrap` to 1.1 in a previous PR, but it looks like it was updated shortly after that PR was merged. I'll update accordingly and as I see them.

**Review by arengoh — APPROVED** [review]

**Review by arengoh — APPROVED** [review]

**Review by arengoh — APPROVED** [review]

**Pull request merged**
